examples/vector_iteration.rs

```diff
@@ -1,14 +1,28 @@
+#[cfg(feature = "serde")]
+extern crate serde;
+
+#[cfg(not(feature = "serde"))]
 extern crate rustc_serialize;
 extern crate mustache;
 
 use std::str;
 use std::collections::HashMap;
 
-#[derive(RustcEncodable)]
+#[cfg_attr(not(feature = "serde"), derive(RustcEncodable))]
 struct User {
     name: String
 }
 
+#[cfg(feature = "serde")]
+impl serde::ser::Serialize for User {
```

Comment:
Yeh, I did actually delete it because I'd made that comment thinking it might affect performance before you pointed out it was actually an example file :) Thanks for the gist! I was under the impression that if you set a capacity explicitly it wouldn't resize until it fills up, like `Vec`.